src/lib.rs

```diff
@@ -603,17 +603,29 @@ pub trait FtdiCommon {
             )
         };
         let (vid, pid) = vid_pid_from_id(device_id);
+        // If the device is not fitted with a configured EEPROM, FT_GetDeviceInfo returns
+        // FT_OTHER_ERROR. In this case, all returned values except for serial_number and
+        // description are valid.
         ft_result(
             DeviceInfo {
                 port_open: true,
                 speed: None,
                 device_type: device_type.into(),
                 product_id: pid,
                 vendor_id: vid,
-                serial_number: slice_into_string(serial_number.as_ref()),
-                description: slice_into_string(description.as_ref()),
+                serial_number: match status {
+                    FT_OK => slice_into_string(serial_number.as_ref()),
+                    _ => String::default(),
+                },
```

Comment:
`slice_to_string` should handle empty strings.

```suggestion
                serial_number: slice_into_string(serial_number.as_ref()),
```